c++/src/kj/async-io.c++

```diff
@@ -750,7 +750,7 @@ private:
           pipe.endState(*this);
         }
 
-        if (actual >= minBytes) {
+        if (actual >= minBytes || actual == 0) {
```

Comment:
This isn't correct, because EOF does not propagate through a pipe. If `tryPumpFrom()` on the write end reaches EOF (of whatever other stream it's pulling from), the write end of the pipe is still open. You could call `tryPumpFrom()` again to pump from a different source, or call write() to push individual buffers of data.

The way EOF would usually propagate is that the promise returned by `tryPumpFrom()` would complete, and then whatever was waiting on that would destroy the write end of the pipe, which then causes the read end to receive EOF.